dprox.js

```diff
@@ -43,12 +43,16 @@ function makeProxy(host, port, hosts) {
 			if(host.preservePrefix) {
 				options.proxyReqPathResolver = req => req.originalUrl;
 			}
-			if(requestHeaders) {
-				options.proxyReqOptDecorator = proxyReqOptions => {
+			options.proxyReqOptDecorator = proxyReqOptions => {
```

Comment:
Does it matter that this now sets `proxyReqOptDecorator` even if it's essentially a no-op, i.e. should we wrap this assignment in `if(requestHeaders || host.ignoreInsecure)`?

Reply:
I used this code with a configuration where one path had neither option set and it seemed to work fine.

Regarding code clarity I get confused by `if(foo||bar) … if(foo) … if(bar) …` nestings (I’ll always double check code containing them because I’m uncertain whether I got it right the first time).

Reply:
I agree that `if` duplication is confusing (irritating even), but it's also more correct?

Having said that, if it works as a no-op, I suppose the only reason to avoid that would be efficiency, which shouldn't be much of a concern here. So I'm good with the way things are.